journals/apps/journals/management/commands/gather_videos.py

```diff
@@ -95,18 +141,49 @@ def get_videos_for_site(self, site):
             })
         return blocks if blocks else []
 
-    def handle(self, *args, **options):
-        """ Collect all videos in courses """
+    def get_videos_for_site(self, site):
+        '''get_videos for given site'''
+        if not hasattr(site, 'siteconfiguration'):
+            self.stderr.write("Missing site config for site {}".format(site))
+            return []
+
+        course_runs = self.get_video_course_runs_for_site(site)
+        return self.get_vidoes_for_course_run(site, course_runs)
 
-        # Contains a list of dicts that contain a course run and all
-        # the blocks in that course run.
-        block_collections = itertools.chain.from_iterable(
-            [self.get_videos_for_site(site) for site in Site.objects.all()]
+    def add_arguments(self, parser):
+        parser.add_argument('--journal_ids', dest='journal_ids', nargs='+', type=int)
+        parser.add_argument('--course_runs', dest='course_runs', nargs='+')
+        parser.add_argument(
+            '--collection_id', dest='collection_id', type=int, help='Collection id to import videos into'
         )
 
+    def handle(self, *args, **options):
+        """ Collect all videos in courses """
+        collection_id = options['collection_id']
```

Comment:
need to initialize video_collections as it's used in line 186 and might not be initialized
`video_collection = None`